Implement login and registration screens in react native

Eduardoespl opened the pull request:
# Description

This pull request introduces the initial implementation of login and registration screens in React Native for our project. Please note that these screens are currently not fully functional but provide the necessary foundation for further development.

## Changes Made

* Created new Login and Register screens using React Native Components.
* Utilized appropiate React Navigation for seamless screen transitions.
* Implemented basic navigation between the login and registration screens.

It's important to highlight that while the screens are visually present and navigable, further functionality and integration with backend APIs are still pending. Future iterations will focus on completing the backend integration for user authentication and account creation.

Please review the code changes thoroughly and provide any feedback or suggestions for improvement.

## Screenshots

![image](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/112664006/2b9b0815-95b5-4e8b-b172-c024c63a8312)

![image](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/112664006/61bfe1b4-ea33-49ec-b0c7-316d3e3580c6)

![Grabación 2023-06-23 174802](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/112664006/57032abe-dcde-4be0-8310-eacad5034976)

Eduardoespl committed:
Implemented registration tab in React Native app with Input component, checkboxes, buttons, and text

Created a new component called "Input" to encapsulate the logic and rendering of text labels and TextInput fields.
Integrated the Input component into the registration tab of the React Native app.
Implemented three instances of the Input component for capturing user input for name, email, and password.
Added two checkboxes to allow users to select optional features or agreements.
Included two buttons, one for registering with user-provided data and another for registering with Google.

Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>
Co-authored-by: arturovaldez019 <arturovaldez019@gmail.com>

Eduardoespl committed:
Create a new view for the login screen, components for the buttons and added navigation between login and register screens.

Created a new login.tsx file and added the code for the login screen.
This file used the input and button components created before.
Also created a new buttons.tsx file and added the code for the buttons.
This buttons get a label prop to set the label of the button and be able to reuse the component.
Added navigation between login and register screens using react navigation.
For this we added a new stack navigator in App.tsx and added screens for login and register, this allow us to call any of this screens from any other screen.

Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>
Co-authored-by: arturovaldez019 <arturovaldez019@gmail.com>

montsegv-2 reviewed (approved):
LGTM!

Alfredo20164920 reviewed (approved):
Good Job 💯

YilKanda commented on AppVuelos/App.tsx:
```diff
@@ -6,113 +6,25 @@
   return (
-    <View style={styles.sectionContainer}>
-      <Text
-        style={[
-          styles.sectionTitle,
-          {
-            color: isDarkMode ? Colors.white : Colors.black,
-          },
-        ]}>
```
The styles go in a separate file.

Eduardoespl replied:
This file don't have styles on the recent commit

YilKanda commented on AppVuelos/components/Input.tsx:
```diff
@@ -0,0 +1,35 @@
+  );
+};
+
+const styles = StyleSheet.create({
+  input: {
+    borderColor: '#9FA9FB',
+    borderWidth: 2,
+    fontSize: 14,
+    fontWeight: '600',
+    marginBottom: 15,
+    marginTop: 5,
+    paddingHorizontal: 10,
+    paddingVertical: 5,
+  },
+  label: {
+    color: '#929292',
+    fontSize: 18,
+    fontWeight: '600',
+  },
+});
```
The styles go in a separate file.

Eduardoespl replied:
Changes made

YilKanda commented on AppVuelos/components/buttons.tsx:
```diff
@@ -0,0 +1,55 @@
+  );
+};
+
+const styles = StyleSheet.create({
+  buttonContainer: {
+    rowGap: 20,
+    marginTop: 20,
+  },
+  rowContainer: {
+    justifyContent: 'center',
+    flexDirection: 'row',
+  },
+  button: {
+    backgroundColor: '#5C6EF8',
+    padding: 12,
+    borderRadius: 10,
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    columnGap: 3,
+  },
+  buttonText: {
+    color: 'white',
+    textAlign: 'center',
+    fontSize: 20,
+    justifyContent: 'space-between',
+  },
+  textNormal: {
+    color: '#929292',
+    textAlign: 'center',
+    fontSize: 16,
+  },
+});
```
The styles go in a separate file.

Eduardoespl replied:
Fixed this problem

YilKanda commented on AppVuelos/screens/login.tsx:
```diff
@@ -0,0 +1,75 @@
+    marginHorizontal: 15,
+  },
+  buttonContainer: {
+    rowGap: 20,
+    marginTop: 20,
+  },
+  title: {
+    color: '#5C6EF8',
+    fontSize: 30,
+    fontWeight: '900',
+  },
+  rowContainer: {
+    marginVertical: 30,
+    justifyContent: 'center',
+    flexDirection: 'row',
+    alignItems: 'center',
+    columnGap: 10,
+  },
+  checkbox: {
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+  },
+  passwordDetails: {
+    color: '#9FA9FB',
+    marginBottom: 20,
+    marginTop: -10,
+  },
+  textLogin: {
+    color: '#929292',
+    textAlign: 'center',
+    fontSize: 16,
+    marginTop: 20,
+  },
+  TextLoginLink: {
+    color: '#5C6EF8',
+    textAlign: 'center',
+    fontSize: 16,
+    marginTop: 20,
+  },
+});
```
The styles go in a separate file.

You can create a specific library for colors.

Eduardoespl replied:
Changes made on screen login.tsx, created a file for the styles and a color library

YilKanda commented on AppVuelos/screens/register.tsx:
```diff
@@ -0,0 +1,104 @@
+    alignItems: 'center',
+  },
+  passwordDetails: {
+    color: '#000',
+    marginBottom: 20,
+    marginTop: -10,
+  },
+  button: {
+    backgroundColor: '#5C6EF8',
+    padding: 12,
+    borderRadius: 10,
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    columnGap: 3,
+  },
+  buttonText: {
+    color: 'white',
+    textAlign: 'center',
+    fontSize: 20,
+    justifyContent: 'space-between',
+  },
+  textNormal: {
+    color: '#929292',
+    textAlign: 'center',
+    fontSize: 16,
+  },
+  textLogin: {
+    color: '#929292',
+    textAlign: 'center',
+    fontSize: 16,
+    marginTop: 20,
+  },
+  TextLoginLink: {
+    color: '#5C6EF8',
+    textAlign: 'center',
+    fontSize: 16,
+    marginTop: 20,
+  },
+});
```
The styles go in a separate file.

You can create a specific library for colors.

Eduardoespl replied:
Changes made

Eduardoespl committed:
Create a new folder for the styles and separate the styles for each component into their own files.

Created a new folder called styles and moved the styles for each component into their own files.
Each file is named after the component it is styling.
Also created a library file for the colors used in the app, to be easily accessible.
To make this changes, I had to change the imports in the components and screens to reflect the new file structure.

Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>
Co-authored-by: arturovaldez019 <arturovaldez019@gmail.com>

toledodavid commented on AppVuelos/screens/login.tsx:
```diff
@@ -0,0 +1,32 @@
+    <SafeAreaView style={loginStyles.container}>
+      <View style={loginStyles.rowContainer}>
+        <Text style={loginStyles.title}>Flight Bookings</Text>
+        <FontAwesome name="plane" size={30} color={'#5C6EF8'} />
```
Probably you should use primaryColor constant instead of hex value directly

cfcortesmx replied:
Done

yovanha commented:
LGTM!

yovanha reviewed (approved):
LGTM!
